src/main/java/uk/gov/hmcts/befta/util/MapVerifier.java

```diff
@@ -226,6 +231,19 @@ private Boolean canAcceptNullFor(Object expectedValue) {
         if (!(expectedValue instanceof String)) {
             return Boolean.FALSE;
         }
-        return !NOT_NULL.equalsIgnoreCase((String) expectedValue);
+        return nullable((String) expectedValue);
+    }
+
+    private boolean nullable(String strExpectedValue) {
+        ExpectedValuePlaceholder expectedValuePlaceholder = ExpectedValuePlaceholder.getByValue(strExpectedValue);
+        return expectedValuePlaceholder != null ?  expectedValuePlaceholder.isNullable(): true;
+    }
+
+    private boolean isValidFormat(String strExpectedValue, Object actualValue) {
```

Comment:
This doesn't sound like logically correct.

Reply:
I didn't understand.. what is the problem in this?